lib/archive/util.go

```diff
@@ -51,20 +51,29 @@ func GenerateFilterSpec(copyPath string, mountPoint string, primaryTarget bool,
 
 func generateCopyFromFilterSpec(copyPath string, mountPoint string, primaryTarget bool) FilterSpec {
 	var filter FilterSpec
-
-	//we only need the right most element.
-	_, first := filepath.Split(copyPath)
+	var first string
+
+	// If the copyPath ends with '/.', we don't want the content's directory name
+	// to be included in the resulting tar header. we do this by removing the directory
+	// portion of the rebase path.
+	// if the copyPath does not end with '/.', we only need the right most element of copyPath.
+	if !strings.HasSuffix(copyPath, "/.") {
```

Comment:
pedantically this should be `filepath.Base(copyPath) == "."`  and I'm not sure why you're then calling `filepath.Split` on it afterwards as `first` will always be `"."` given the input.